lib/mongo/operation/context.rb

```diff
@@ -58,17 +59,42 @@ def initialize(
 
         @client = client
         @session = session
+        @view = view
         @connection_global_id = connection_global_id
         @deadline = calculate_deadline(operation_timeouts, session)
+        @operation_timeouts = operation_timeouts
         @timeout_sec = if @deadline then @deadline - Utils.monotonic_time end
         @options = options
       end
 
       attr_reader :client
       attr_reader :session
+      attr_reader :view
       attr_reader :deadline
       attr_reader :timeout_sec
       attr_reader :options
+      attr_reader :operation_timeouts
+
+      # Returns a new Operation::Context with the deadline refreshed
+      # and relative to the current moment.
+      #
+      # @return [ Operation::Context ] the refreshed context
+      def refresh(connection_global_id: @connection_global_id, timeout_ms: nil, view: nil)
+        operation_timeouts = @operation_timeouts
+        operation_timeouts = operation_timeouts.merge(operation_timeout_ms: timeout_ms) if timeout_ms
+
+        self.class.new(client: client,
+                       session: session,
+                       connection_global_id: connection_global_id,
+                       operation_timeouts: operation_timeouts,
+                       view: view || self.view,
+                       options: options)
+      end
+
+      def timeout_ms
+        @operation_timeouts[:inherited_timeout_ms] ||
```

Comment:
Should the `operation_timeout_ms` have the higher priority?

Reply:
good catch! 👍

Reply:
Actually, I don't think that method is even used anywhere...I think it was an artifact from an earlier experiment. I'll remove it.